certbot-ci/certbot_integration_tests/utils/certbot_call.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+from __future__ import absolute_import
+from distutils.version import LooseVersion
+import subprocess
+import sys
+import os
+
+from certbot_integration_tests.utils import misc
+
+
+def certbot_test(certbot_args, directory_url, http_01_port, tls_alpn_01_port,
+                 config_dir, workspace, force_renew, capture_stdout):
+    new_environ = os.environ.copy()
+    new_environ['TMPDIR'] = workspace
+
+    additional_args = []
+    if misc.get_certbot_version() >= LooseVersion('0.30.0'):
+        additional_args.append('--no-random-sleep-on-renew')
+
+    if force_renew:
+        additional_args.append('--renew-by-default')
+
+    command = [
+        'certbot',
+        '--server', directory_url,
+        '--no-verify-ssl',
+        '--http-01-port', str(http_01_port),
+        '--https-port', str(tls_alpn_01_port),
+        '--manual-public-ip-logging-ok',
+        '--config-dir', config_dir,
+        '--work-dir', os.path.join(workspace, 'work'),
+        '--logs-dir', os.path.join(workspace, 'logs'),
+        '--non-interactive',
+        '--no-redirect',
+        '--agree-tos',
+        '--register-unsafely-without-email',
+        '--debug',
+        '-vv'
+    ]
+
+    command.extend(certbot_args)
+    command.extend(additional_args)
+
+    subprocess_call = subprocess.check_output if capture_stdout else subprocess.check_call
+
+    print('--> Invoke command:\n=====\n{0}\n====='.format(subprocess.list2cmdline(command)))
+    return subprocess_call(command, universal_newlines=True,
```

Comment:
I admit this can be really different values. Ideally, I would like this function to do both: returning the output, and piping it to standard stdout. Because first case to cover here is from certbot tests, where the returned output need to be asserted, and second case is call from shell, where you expect to see dynamically the output of the command.

However, this is hard to code. I mean, you can capture everything, then print the output before returning it, but then you see the all output in one bunch only at the end of the command. Not great for a shell command. 

Note that it would be also interesting during the tests themselves. Currently, pytest displays correctly the output if a test fails, because in fact all output is redirected in stderr, that is not captured by subprocess, but directly piped to stderr, so captured by pytest. 

All of this to say:
1) We can keep things like this
2) We can have two dedicated methods, one using `subprocess.check_call, another using `subprocess.check_output` and call appropriately
3) Implement the logic to do everything in the same method (returning output and piping it dynamically to stdout)
4) Your idea?

Reply:
Done, with option 2. In fact, I remove the specific branch of calling certbot without output capturing, to put it directly in the `main` function, since this is the only place where is is useful.